assets/js/translator.js

```diff
@@ -730,6 +744,115 @@ function translateDoc() {
+    // testing – but on the other hand most uses of document.write are evil.
+    return html.replace(/document[.]write[(]/g,
+        'console.log("document.write "+');
+}
+
+function translateWebpage() {
+    persistChoices('translator', true);
+    if(!$('div#translateWebpage').is(':visible')) {
+        showTranslateWebpageInterface($('#originalText').val().trim());
+    }
+
+    if(pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1) {
+        sendEvent('translator', 'translateWebpage', curSrcLang + '-' + curDstLang);
+        $('iframe#translatedWebpage').animate({'opacity': 0.75}, 'fast');
+        textTranslateRequest = callApy({
+            data: {
+                'langpair': curSrcLang + '|' + curDstLang,
+                'markUnknown': 'no',
+                'url': $('input#webpage').val()
+            },
+            dataType: 'json',
+            success: handleTranslateWebpageSuccessResponse,
+            error: handleTranslateWebpageErrorResponse,
+            complete: function () {
+                ajaxComplete();
+                textTranslateRequest = undefined;
+                $('iframe#translatedWebpage').animate({'opacity': 1}, 'fast');
+            }
+        }, '/translatePage', true);
+    }
+}
+
+function handleTranslateWebpageSuccessResponse(data) {
+    if(data.responseStatus === HTTP_OK_CODE) {
+        var iframe = $('<iframe id="translatedWebpage" class="translatedWebpage" frameborder="0"></iframe>')[0];
+        $('#translatedWebpage').replaceWith(iframe);
+        iframe.contentWindow.document.open();
+        var html = cleanPage(data.responseData.translatedText);
+        iframe.contentWindow.document.write(html);
+        iframe.contentWindow.document.close();
+        var contents = $(iframe).contents();
+        contents.find('head')
+            .append($('<base>').attr('href', $('input#webpage').val()));
+        $(iframe).load(function () {
+            $.each(contents.find('a'), function (index, a) {
+                var href = a.href;
+                $(a).on('click', function () { window.parent.translateLink(href); });
+                a.href = '#';
+                a.target = '';
+            });
+        });
+    }
+    else {
+        translationNotAvailableWebpage(data);
+    }
+}
+
+function handleTranslateWebpageErrorResponse(jqXHR) {
+    translationNotAvailableWebpage(jqXHR.responseJSON);
+}
+
+function showTranslateWebpageInterface(url) {
+    $('#srcLangSelectors').css({
+        'padding-left': '5px',
+        'width': 'calc(87% - 34px)'
+    });
+    $('div#translateText').fadeOut('fast', function () {
+        $('input#webpage').attr({
+            'required': true,
+            'novalidate': false
+        });
+        $('button#cancelWebpageTranslate').fadeIn('fast');
+        $('div#translateWebpage').fadeIn('fast');
+    });
+    if(url && typeof url === 'string') {
```

Comment:
I think the cleaner way of doing this is running it without an argument on translateWebpage click so you don't need to check for type here.